src/cmds/qsub.c

```diff
@@ -472,15 +472,27 @@ expand_varlist(char *varlist)
 	char *p1, *p2, *p;
 	char *ev;
 	int  v_value1_sz=0;
+	char *pc;
+	int special_char_cnt = 0;
+
+	/*
+	 * count special characters as they are escaped with '\' in copy_env_value function
+	 * so that this is useful while calculating the accurate size of the destination string
+	 */
+	pc = varlist;
+	for(;*pc;pc++) {
+		if ((*pc == '"') || (*pc == '\'') || (*pc == ',') || (*pc == '\\'))
+			special_char_cnt++;
+	}
```

Comment:
If the varlist is large, it will slow down qsub if we are running over it to find the special characters. Why not simply assume that we will need double of the stringlength?

Reply:
Since I have implemented your second comment where in strlen() is avoided and we scan the string only once for calculating both the length of the string as well as special character count, hope this code is fine now.

Reply:
That looks better, thansk